public/app/plugins/datasource/loki/querybuilder/operationUtils.ts

```diff
@@ -148,7 +148,7 @@ function operationWithRangeVectorRenderer(
 }
 
 export function labelFilterRenderer(model: QueryBuilderOperation, def: QueryBuilderOperationDef, innerExpr: string) {
-  if (model.params[1] === '<' || model.params[1] === '>') {
+  if (!Number.isNaN(Number(model.params[2]))) {
```

Comment:
We want to decide about this based on `operator`, not based on value. So we just want to check for following operators: `<`, `<=`, `>=` `>`. 

This is because for all other operators, such as `=`, `=~`, ... the query should include string, not number.

Reply:
@svennergr tried it in here https://grafana.com/docs/loki/latest/logql/analyzer/?query=%7C+logfmt+%7C+duration+%3E+9s&line%5B%5D=level%3Dinfo+ts%3D2022-03-23T11%3A55%3A29.846163306Z+caller%3Dmain.go%3A112+msg%3D%22Starting+Grafana+Enterprise+Logs%22+duration%3D10s and comparing durations (and probably bytes as well) works.